Mathlib/CategoryTheory/Limits/Preserves/Finite.lean

```diff
@@ -3,8 +3,8 @@ Copyright (c) 2021 Andrew Yang. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Andrew Yang
 -/
-import Mathlib.CategoryTheory.Limits.Preserves.Basic
 import Mathlib.CategoryTheory.Limits.Shapes.FiniteProducts
+import Mathlib.CategoryTheory.Limits.FunctorCategory.Basic
 
```

Comment:
Rather than changing the imports here, you should put the declarations that use material from `FunctorCategory.Basic` in either the file `CategoryTheory.Limits.FunctorCategory.Finite` or `CategoryTheory.Limits.Preserves.FunctorCategory`.

Reply:
Neither of these files work, because I'd need to import `Mathlib.CategoryTheory.Limits.Preserves.Finite` in them and that causes a cycle.

Reply:
Ok, then I would suggest creating a new file `CategoryTheory.Limits.Preserves.FunctorCategory.Finite`, and moving `Preserves.FunctorCategory` to `Preserves.FunctorCategory.Basic`

Reply:
Hmm, the file called `Preserves.FunctorCategory` right now is not particularly basic. So it should probably not be called basic but something else.

Reply:
I think adding the import `Preserves.Finite` to `Preserves.FunctorCategory` should work as well

Reply:
If you do this just make sure to update the module docstring in that file

Reply:
Somehow removing two unnecessary imports made my comment above obsolete and I now put it into `ategoryTheory.Limits.FunctorCategory.Finite`, is that ok?